bin/version-include.sh

```diff
@@ -0,0 +1,104 @@
+# Copyright © 2021, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+function populateValuesYAML() {
+  v4mValuesYAML=$1
+  rm -f "$v4mValuesYAML"
+  touch "$v4mValuesYAML"
+
+  # Attempt to obtain current git commit hash
+  gitCommit=$(git rev-parse --short HEAD 2>/dev/null)
+  if [ -n "$gitCommit" ]; then
+    echo "gitCommit: $gitCommit" >> "$v4mValuesYAML"
+  fi
+
+  # List contents of USER_DIR
+  if [ -d "$USER_DIR" ]; then
+    echo '"user_dir":' >> "$v4mValuesYAML"
+    echo "  path: $(dirname $USER_DIR)" >> "$v4mValuesYAML"
+    echo '  files: |' >> "$v4mValuesYAML"
+    l=($(find "$USER_DIR" -type f))
+    for (( i=0; i<${#l[@]}; i++ )); do
+      fullPath=${l[i]}
+      path=${fullPath#"$USER_DIR/"}
+      echo "      $path" >> "$v4mValuesYAML"
+    done
+  fi  
+  # Top-level user.env contents
+  if [ -f "$USER_DIR/user.env" ]; then
+    echo '  "user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+  # Monitoring user.env contents
+  if [ -f "$USER_DIR/monitoring/user.env" ]; then
+    echo '  "monitoring_user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/monitoring/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+  # Logging user.env contents
+  if [ -f "$USER_DIR/logging/user.env" ]; then
+    echo '  "logging_user.env": |' >> "$v4mValuesYAML"
+    cat "$USER_DIR/logging/user.env" | sed 's/^/      /' >> "$v4mValuesYAML"
+  fi
+}
+
+function deployV4MInfo() {
+  NS=$1
+  if [ -z "$NS" ]; then
+    log_error "No namespace specified for deploying version info"
+    return 1
+  fi
```

Comment:
I don't see any logic in the calling programs to deal with a non-zero return code.  Do we want them to error out at that point?  Generate a message?  Do nothing?

Reply:
Fixed